src/main.rs

```
@@ -54,15 +54,15 @@ async fn main() -> io::Result<()> {
     init_logging(&log_args);
 
     if let Err(err) = run(command).await.as_ref() {
-        tracing::error!(error = ?err, "network tunnel failed.");
+        tracing::error!(error = %err, "network tunnel failed.");
```

% signifies the `Display` implementation of the error, rather than its `Debug` implementation, so the error looks more human-readable rather than exposing the underlying error struct

---

This makes sense and is fine for `std::io::Result` :+1: But a general informational comment:

We use a lot of `anyhow::Result` which is a really nice crate for chaining errors with contextual information and back traces, and I generally recommend it for all top-level handling of errors.

For anyhow::Result, we _always_ want to use the `?err` form when used in a tracing::* message, because that renders the complete error chain in a human-readable fashion. We've previously had situations where we were using the `%err` form accidentally, which only renders the top-most error context of the chain, and then had to go back and switch to `?err` as part of an investigation because the meat of the error was lost.